osu.Game/Database/BeatmapDatabase.cs

```diff
@@ -118,12 +118,43 @@ public void Reset()
 
         public void Import(IEnumerable<string> paths)
         {
+            Stack<BeatmapSetInfo> sets = new Stack<BeatmapSetInfo>();
+
             foreach (string p in paths)
-                Import(p);
+                try
+                {
+                    BeatmapSetInfo set = importBeatmapSet(p);
+
+                    sets.Push(set);
+
+                    // We may or may not want to delete the file depending on where it is stored.
+                    //  e.g. reconstructing/repairing database with beatmaps from default storage.
+                    // TODO: Add a check to prevent files from storage to be deleted.
+                    try
+                    {
+                        File.Delete(p);
+                    }
+                    catch (Exception e)
+                    {
+                        Logger.Error(e, $@"Could not delete file at {p}");
+                    }
+                }
+                catch (Exception e)
+                {
+                    Logger.Error(e, $@"Could not import beatmap set");
+                }
+            
+            // Batch commit with multiple sets to database
+            Import(sets);
         }
 
         public void Import(string path)
         {
+            Import(new [] { path });
+        }
+
+        private BeatmapSetInfo importBeatmapSet(string path)

```

Comment:
weird name as this doesn't actually perform the database import (just copying of file and checking for pre-existing). should also document the null return on already-existing.